tests/x11/user_defined_snapshot.pm

```diff
@@ -0,0 +1,100 @@
+# SUSE's openQA tests
+#
+# Copyright © 2012-2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test that zypper can refresh https repositories
+# Maintainer: Stephan Kulow <coolo@suse.com>
+
+use base "x11test";
+use strict;
+use testapi;
+use utils qw/sle_version_at_least/;
+
+sub handle_login {
+    if (get_var('DESKTOP_MINIMALX_INSTONLY')) {
+        # return at the DM and log in later into desired wm
+        return;
+    }
+    mouse_hide();
+    wait_still_screen;
+    if (get_var('DM_NEEDS_USERNAME')) {
+        type_string $username;
+    }
+    if (match_has_tag("sddm")) {
+        # make sure choose plasma5 session
+        assert_and_click "sddm-sessions-list";
+        assert_and_click "sddm-sessions-plasma5";
+        assert_and_click "sddm-password-input";
+    }
+    else {
+        send_key "ret";
+        if (!check_screen "displaymanager-password-prompt") {
+            record_soft_failure;
+            assert_screen "displaymanager-password-prompt";
+        }
+    }
+    type_string "$password";
+    send_key "ret";
+}
+
+sub y2snapper_create_snapshot() {
+    my $self = shift;
+    my $name = shift || "grub_comment";
+    my $user_data = shift || 'bootloader="Bootloader_Comment"';
+    # Open the 'C'reate dialog and wait until it is there
+    send_key "alt-c";
+    assert_screen 'yast2_snapper-createsnapshotdialog', 100;
+    # Fill the form and finish by pressing the 'O'k-button
+    type_string $name;
+    send_key "alt-u"; # match User data column
+    type_string $user_data;
+    save_screenshot;
+    send_key "alt-o";
+    save_screenshot;
+}
+
+sub run() {
+    my $self = shift;
+    # Start an xterm as root
+    x11_start_program("xterm");
+    wait_idle;
+    become_root;
+    script_run "cd";
+
+    # Start the yast2 snapper module and wait until it is started
+    type_string "yast2 snapper\n";
+    assert_screen 'yast2_snapper-snapshots', 100;
```

Comment:
To put this in perspective: copy&paste is fine while drawing up the scenario - but when you're done with your proof of concept, you need to take effort to de-duplicate the code.